core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -596,7 +597,9 @@ private[spark] class TaskSetManager(
     removeRunningTask(tid)
     info.markFailed()
     val index = info.index
-    copiesRunning(index) -= 1
+    if (copiesRunning(index) >= 1) {
```

Comment:
Believe this is a bad artifact of me playing with things to try to get tests to pass. Please try to revert this change.